src/Security/Member.php

```diff
@@ -1702,4 +1711,52 @@ public function getHtmlEditorConfigForCMS()
         // If can't find a suitable editor, just default to cms
         return $currentName ? $currentName : 'cms';
     }
+
+    /**
+     * This needs to be public because it's referenced in getMemberPasswordField() and called in ConfirmedPasswordField
+     */
+    public static function generateRandomPassword(array $args = []): string
+    {
+        $password = '';
+        $validator = self::password_validator();
+        $minLength = $validator ? $validator->getMinLength() : 20;
+        $maxLength = $args['maxLength'] ?? 0;
+        if ($maxLength === 0) {
+            $maxLength = max($minLength, 20);
+        } else {
+            if ($maxLength < $minLength) {
+                throw new InvalidArgumentException('maxLength argument is less than PasswordValidator minLength');
+            }
+        }
+        // Create a dummy member to test the password validator against
+        // Use a dummy member because this is intended to be used in a context where an Administrator
+        // is created a branch new member who doesn't yet have an ID
+        $dummyMember = Member::create();
```

Comment:
Since we already know what the validator's validation logic is, we don't need to call validate against a dummy member, right? We just need to make a valid password the first time around.

Reply:
Method signature (using docblocks) is PasswordValidator::validate(string $password, Member $member).  We need the member if we want to check against the PasswordValidator

Reply:
Have reworked logic to make method non-static to allow for using $this